tools/accuracy_checker/accuracy_checker/metrics/reid.py

```diff
@@ -339,45 +339,70 @@ class NormalizedEmbeddingAccuracy(FullDatasetEvaluationMetric):
-            camera_id = ann.camera_id
-
-            for j, gallery_embedding in enumerate(gallery_embeddings):
-                gallery_person_id = annotations[j].person_id
-                gallery_camera_id = annotations[j].camera_id
+    def configure(self):
+        self.top_k = self.get_value_from_config('top_k')
 
-                if person_id == gallery_person_id and camera_id == gallery_camera_id:
-                    continue
+    @classmethod
+    def parameters(cls):
+        parameters = super().parameters()
+        parameters.update({
+            'top_k': NumberField(
+                value_type=int, min_value=1, default=1, optional=True,
+                description="Number of k highest ranked samples to consider when matching."
+            )
+        })
+        return parameters
 
-                normalized_dot = np.linalg.norm(gallery_embedding) * np.linalg.norm(pred.embedding)
-                sim = np.dot(gallery_embedding, pred.embedding) / normalized_dot
+    @staticmethod
+    def extract_person_id(annotations, query=False):
+        return np.array([a.person_id for a in annotations if a.query == query])
 
-                if best_sim < sim:
-                    best_sim = sim
-                    pred_person_id = gallery_person_id
+    @staticmethod
+    def extract_cam_id(annotations, query=False):
+        return np.array([a.camera_id for a in annotations if a.query == query])
 
-            if pred_person_id == ann.person_id:
-                true_positive += 1
-            else:
-                false_positive += 1
+    @staticmethod
+    def eval_valid_matrix(gallery_person_ids, gallery_cam_ids, query_person_ids, query_cam_ids):
+        person_id_mask = np.tile(gallery_person_ids, (len(query_person_ids), 1))
+        person_id_mask = (person_id_mask == np.tile(query_person_ids,
+                                                    (len(gallery_person_ids), 1)).T)
+        cam_id_mask = np.tile(gallery_cam_ids, (len(query_cam_ids), 1))
+        cam_id_mask = (cam_id_mask == np.tile(query_cam_ids, (len(gallery_cam_ids), 1)).T)
+        return 1 - person_id_mask*cam_id_mask
 
-        if (true_positive + false_positive) == 0:
-            return [0]
 
-        accuracy = true_positive / (true_positive + false_positive)
-        return [accuracy]
+    def evaluate(self, annotations, predictions):
+        gallery_embeddings = extract_embeddings(
+            annotations, predictions, query=False)
+        gallery_person_ids = self.extract_person_id(annotations, False)
+        gallery_cam_ids = self.extract_cam_id(annotations, False)
+        query_embeddings = extract_embeddings(
+            annotations, predictions, query=True)
+        query_person_ids = self.extract_person_id(annotations, True)
+        query_cam_ids = self.extract_cam_id(annotations, True)
+
+        valid_mask = self.eval_valid_matrix(
+            gallery_person_ids, gallery_cam_ids, query_person_ids, query_cam_ids)
```

Comment:
@tlsdmstn56-2 we accept standard line length - 120 symbols, no need to move in new line if your code pass to this limit (on my view, it makes code less readable)
```suggestion
        valid_mask = self.eval_valid_matrix(gallery_person_ids, gallery_cam_ids, query_person_ids, query_cam_ids)
```